pkg/scaffold/project.go

```diff
@@ -16,13 +19,103 @@ type ProjectTemplate struct {
 	StackTemplates []*StackTemplate `json:"stacks,omitempty" yaml:"stacks,omitempty"`
 }
 
+// StackTemplates contains configuration in stack level
 type StackTemplate struct {
 	// Name is stack name
 	Name string `json:"name" yaml:"name"`
-	// Fields contains all fields wait to be prompt
+	// Fields contains all stack fields definition
 	Fields []*FieldTemplate `json:"fields,omitempty" yaml:"fields,omitempty"`
 }
 
+// FieldTemplate can describe all kinds of type, including primitive and composite.
+// for example:
+//
+// String:
+//
+//  strField := FieldTemplate{
+//    Name:    "string field",
+//    Type:    StringField,
+//    Default: "bar",
+//  }
+//
+// Array, []int:
+//
+//  arrayField := &FieldTemplate{
+//    Name: "array field",
+//    Type: ArrayField,
+//    Elem: &FieldTemplate{
+//      Type: IntField,
+//    },
+//    Default: []int{1, 2, 3},
+//  }
+//
+// Map, map[string]bool:
+//
+//  mapField := &FieldTemplate{
+//    Name: "map field",
+//    Type: MapField,
+//    Key: &FieldTemplate{
+//      Type: StringField,
+//    },
+//    Value: &FieldTemplate{
+//      Type: BoolField,
+//    },
+//    Default: map[string]bool{
+//      "foo": true,
+//      "bar": false,
+//    },
+//  }
+//
+// Struct, including primitive, composite type and nested struct:
+//
+//  structField := &FieldTemplate{
+//    Name: "struct field",
+//    Type: StructField,
+//    Fields: []*FieldTemplate{
+//      {
+//        Name: "float field",
+//        Type: FloatField,
+//      },
+//      {
+//        Name: "array field",
+//        Type: ArrayField,
+//        Elem: &FieldTemplate{
+//          Type: IntField,
+//        },
+//      },
+//      {
+//        Name: "map field",
+//        Type: MapField,
+//        Key: &FieldTemplate{
+//          Type: StringField,
+//        },
+//        Value: &FieldTemplate{
+//          Type: BoolField,
+//        },
+//      },
+//      {
+//        Name: "inner struct",
+//        Type: StructField,
+//        Fields: []*FieldTemplate{
+//          {
+//           Name: "foo",
+//           Type: StringField,
+//          },
+//        },
+//      },
+//    },
+//    Default: map[string]interface{}{
+//      "string field": "foo",
+//      "array field":  []int{1, 2, 3},
+//      "map field": map[string]bool{
+//        "foo": true,
+//        "bar": false,
+//      },
+//      "inner struct": map[string]interface{}{
+//        "foo": "bar",
+//      },
+//    },
+//  }
```

Comment:
the doc code is too large, put it into the example test code

Reply:
ok, thanks for your advice